libr/bin/p/bin_xtr_fatmach0.c

```diff
@@ -92,7 +92,14 @@ static RBinXtrData * extract(RBin* bin, int idx) {
 	arch = r_bin_fatmach0_extract (fb, idx, &narch);
 	if (!arch) return res;
 
-	res = r_bin_xtrdata_new (NULL, NULL, arch->b, arch->offset, arch->size, narch);
+	RBinXtrMetadata *metadata = R_NEW0 (RBinXtrMetadata);
+
+	metadata->bits = MACH0_(get_bits) (r_buf_buffer(arch->b));
```

Comment:
cache r_buf_buffer before and cast in MACH0_(get_bits) to avoid warning when compiling. There is still a lot take a look those please. The segfault in fatmach is because metadata->arch has a value that is not mapped.

```
==82007== Invalid read of size 1
==82007==    at 0x1000296BF: strlen (in /usr/local/Cellar/valgrind/3.11.0/lib/valgrind/vgpreload_memcheck-amd64-darwin.so)
==82007==    by 0x100005062: extract_binobj (in /usr/local/bin/rabin2)
==82007==    by 0x10000467F: rabin_extract (in /usr/local/bin/rabin2)
==82007==    by 0x100003FFF: main (in /usr/local/bin/rabin2)
==82007==  Address 0x1c90b70 is not stack'd, malloc'd or (recently) free'd
==82007==
==82007==
==82007== Process terminating with default action of signal 11 (SIGSEGV)
==82007==  Access not within mapped region at address 0x1C90B70
==82007==    at 0x1000296BF: strlen (in /usr/local/Cellar/valgrind/3.11.0/lib/valgrind/vgpreload_memcheck-amd64-darwin.so)
==82007==    by 0x100005062: extract_binobj (in /usr/local/bin/rabin2)
==82007==    by 0x10000467F: rabin_extract (in /usr/local/bin/rabin2)
==82007==    by 0x100003FFF: main (in /usr/local/bin/rabin2)
```